### PR TITLE
Add docstrings for labeling module

### PR DESCRIPTION
**Pull request by henryre**
Add docstrings for most of `snorkel/labeling`. Per today's discussion, using NumPy style with PEP 484. Uncovered paths are `snorkel/labeling/model` and `snorkel/labeling/analysis.py`.

**Test plan**
`pydocstyle snorkel/labeling`

### CODE REVIEW

**Comment by henryre** [comment]
**Changes planned**: going to add some color on what an LF actually is

File: snorkel/labeling/lf.py [hunk 1]
```diff
@@ -6,6 +6,53 @@
...
+    decorator. See `labeling_function`.
+
+    Parameters
+    ----------
+    name
+        Name of the LF
+    f
+        Function that implements the core LF logic
+    label_space
+        Set of labels the LF can output, including 0. By default None.
```
Comment (bhancock8) [inline snorkel/labeling/lf.py]: We've got a lot of "by default None" statements in the docstrings right now, and that's already specified by the explicit typing. I'd vote to drop all these?

File: snorkel/labeling/lf.py [hunk 2]
```diff
@@ -66,6 +132,39 @@ def __repr__(self) -> str:
...
 class labeling_function:
+    """Decorator to define a LabelingFunction object from a function.
+
+    Parameters
+    ----------
+    name
+        Name of the LF, by default None which uses the name of the
+        wrapped function.
+    label_space
+        Set of labels the LF can output, including 0. By default None.
```
Comment (bhancock8) [inline snorkel/labeling/lf.py]: Same--bunch of "by default None"s

File: snorkel/labeling/preprocess/nlp.py [hunk 3]
```diff
@@ -22,4 +43,16 @@ def __init__(
...
+
+        Parameters
+        ----------
+        text
+            Text of document to parse
+
+        Returns
+        -------
+        FieldMap
+            Mapping containing parsed document
```
Comment (bhancock8) [inline snorkel/labeling/preprocess/nlp.py]: "Mapping containing parsed document"  doesn't tell me what's being mapped to what. The key names are spacy attribute names and the values are the values of the tokens for that attribute, right?

File: snorkel/labeling/lf.py [hunk 4]
```diff
@@ -99,6 +182,18 @@ def g(x: DataPoint) -> int:
         self.fault_tolerant = fault_tolerant
 
     def __call__(self, f: Callable[..., int]) -> LabelingFunction:
+        """Wrap a function to create a `LabelingFunction`.
```
Comment (paroma) [inline snorkel/labeling/lf.py]: `LabelingFunction` vs. `LabelFunction` (under returns)

File: snorkel/labeling/lf.py [hunk 5]
```diff
@@ -36,9 +72,23 @@ def __init__(
         self._preprocessors = preprocessors or []
 
     def set_fault_tolerant(self, fault_tolerant: bool = True) -> None:
+        """Change LF fault tolerance mode.
```
Comment (paroma) [inline snorkel/labeling/lf.py]: little more detail?

**Comment by henryre** [comment]
@bhancock8 @paroma changes made

**Pull request merged**